### PR TITLE
add aggregations to ISearchResult

### PR DESCRIPTION
**Pull request by pranavkulkarni**

### CODE REVIEW

File: packages/arcgis-rest-portal/src/util/search.ts [hunk 1]
```diff
@@ -23,4 +23,5 @@ export interface ISearchResult<T extends IItem | IGroup> {
   nextStart: number;
   results: T[];
   nextPage?: () => Promise<ISearchResult<T>>;
+  aggregations?: { counts: any };
```
Comment (jgravois) [inline packages/arcgis-rest-portal/src/util/search.ts]: aggregations aren't very self-explanatory. lets add the doc below (and denote that counts will be an array).

```ts
/**
 * Aggregations will only be present if [`fieldCounts`](https://developers.arcgis.com/rest/users-groups-and-items/search.htm#GUID-1C625F8A-4A12-45BB-B537-74C82315759A) are requested.
 */
aggregations?: { counts: any[] };
```

Comment (pranavkulkarni) [reply packages/arcgis-rest-portal/src/util/search.ts]: cool thanks

Comment (patrickarlt) [reply packages/arcgis-rest-portal/src/util/search.ts]: @jgravois do you think we should type `counts` more strongly?

Comment (jgravois) [reply packages/arcgis-rest-portal/src/util/search.ts]: at first glance i thought the payload was too complicated to bother, but it's actually not.

File: packages/arcgis-rest-portal/src/util/search.ts [hunk 2]
```diff
@@ -23,4 +23,5 @@ export interface ISearchResult<T extends IItem | IGroup> {
   nextStart: number;
   results: T[];
   nextPage?: () => Promise<ISearchResult<T>>;
+  aggregations?: { counts: any[] };
```
Comment (jgravois) [inline packages/arcgis-rest-portal/src/util/search.ts]:
```suggestion
  aggregations?: { counts: {
    fieldName: string;
    fieldValues: {
      value: any;
      count: number;
    }[]
  } };
```

Comment (patrickarlt) [reply packages/arcgis-rest-portal/src/util/search.ts]: Looks good but I don't think this pass lint.

Comment (pranavkulkarni) [reply packages/arcgis-rest-portal/src/util/search.ts]: Thanks guys

**Pull request merged**